firestore/transaction.go

```diff
@@ -59,7 +62,29 @@ var ReadOnly = ro{}
 
 type ro struct{}
 
-func (ro) config(t *Transaction) { t.readOnly = true }
+func (ro) config(t *Transaction)                     { t.readOnly = true }
+func (ro) handleCommitResponse(r *pb.CommitResponse) {}
+
+// Wrapper over pb.CommitResponse to avoid exposing proto directly. This serves
+// both as the option (from WithCommitResponseTo, below) and the wrapper.
+type CommitResponse struct {
+	response *pb.CommitResponse
+}
+
+func (cr *CommitResponse) config(t *Transaction) {}
+func (cr *CommitResponse) handleCommitResponse(r *pb.CommitResponse) {
+	cr.response = r
+}
+
+func (cr *CommitResponse) CommitTime() time.Time {
+	return cr.response.CommitTime.AsTime()
+}
+
+// WithCommitResponseTo is a TransactionOption that allows the caller to indicate where the commit
+// response for the transaction should be stored, upon successful commit.
```

Comment:
The API is a bit confusing so provide a small example in the doc:

//    var cr firestore.CommitResponse
//    err := client.RunTransaction(...., firestore.WithCommitResponseTo(&cr))
//    if err != nil {...}
//    // CommitReponse can be accessed here.

Actually one thing I'm not clear about is, is it populated if the transaction fails? Can that be documented too?

Reply:
It's only populated on success -- will document:

```
// on success, handle the commit response
if err == nil {
	for _, opt := range opts {
		opt.handleCommitResponse(commitResponse)
	}
}
```

Reply:
Done.

Reply:
The example needs to be added to firestore/examples_test.go 
I think thats what @jba meant.

Rest of the changes LGTM.

Once @jba has approved, I'll approve it too

Reply:
@bhshkh @bja Gotcha, my mistake. I can make that change. 

Though I notice that, now that I'm returning an unexported ```commitResponse``` from ```WithCommitResponseTo```, one of the checks is failing with:

> firestore/transaction.go:96:46: exported func WithCommitResponseTo returns unexported type firestore.commitResponse, which can be annoying to use

I can think of a couple solutions.
* Change ```commitResponse``` to ```CommitResponseOption``` (we're already using ```CommitResponse``` for the wrapper struct)
* Return ```TransactionOption``` from ```WithCommitResponseTo```

In the spirit of avoiding back-and-forths, do either of you have a preference here (or alternate suggestion)?

Reply:
I would wait for @jba's review.

Reply:
@jba I was just wondering if you prefer either option or have a better idea. I can make the changes today. Thanks.

Reply:
@jba @bhshkh Just checking in on this one. If nobody has a better suggestion, I guess I'll go with the first option, changing the return type to ```CommitResponseOption```.

Reply:
@jba @bhshkh I updated the example, renamed the option, and rebased to main. Please let me know if that looks ok. Thanks.